packages/python/plotly/plotly/tests/test_core/test_px/test_px.py

```diff
@@ -51,3 +51,111 @@ def test_custom_data_scatter():
         fig.data[0].hovertemplate
         == "sepal_width=%{x}<br>sepal_length=%{y}<br>petal_length=%{customdata[2]}<br>petal_width=%{customdata[3]}<br>species_id=%{customdata[0]}"
     )
+
+
+def test_px_templates():
+    import plotly.io as pio
+    import plotly.graph_objects as go
+
+    tips = px.data.tips()
+
+    # use the normal defaults
+    fig = px.scatter()
+    assert fig.layout.template == pio.templates[pio.templates.default]
+
+    # respect changes to defaults
+    pio.templates.default = "seaborn"
+    fig = px.scatter()
+    assert fig.layout.template == pio.templates["seaborn"]
+
+    # special px-level defaults over pio defaults
+    pio.templates.default = "seaborn"
+    px.defaults.template = "ggplot2"
```

Comment:
we should document this in https://plot.ly/python/templates/, I did not know this was possible